### PR TITLE
Logging: Add logging configuration

### LINKED ISSUE #1097: How to configure loglevels
Looking at NestJS they support the debug levels
```
'log' | 'error' | 'warn' | 'debug' | 'verbose'
```

which can be individually included. Like so:
```
const app = await NestFactory.create(AppModule, {
  logger: ['error', 'warn'],
});
```
or
```
const app = await NestFactory.create(AppModule);
app.useLogger(['error', 'warn']);
```

Our backend currently defines these log levels:
```
export enum Loglevel {
  TRACE = 'trace',
  DEBUG = 'debug',
  INFO = 'info',
  WARN = 'warn',
  ERROR = 'error',
}
```

The intent with these loglevels was that they are additive. So I'd suggest the following configuration
```
our loglevel -> nest log level config
ERROR -> ['error']
WARN -> ['error', 'warn']
INFO -> ['error', 'warn', 'log']
DEBUG -> ['error', 'warn', 'log', 'debug']
TRACE -> ['error', 'warn', 'log', 'debug', 'verbose']
```

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
logging

### Description
This PR add the functionality to configure which log messages should be shown on the basis of the appConfigs loglevel.

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### Related Issue(s)
fixes #1097

### CODE REVIEW

**Review by InnayTool — APPROVED** [review]

**Review by InnayTool — DISMISSED** [review]

**Review by InnayTool — DISMISSED** [review]

File: src/logger/console-logger.service.ts [hunk 1]
```diff
@@ -29,43 +35,63 @@ export class ConsoleLoggerService {
...
 
   log(message: unknown, functionContext?: string): void {
-    this.printMessage(
-      message,
-      clc.green,
-      this.makeContextString(functionContext),
-      false,
-    );
+    if (
+      this.appConfig.loglevel === Loglevel.INFO ||
```
Comment (davidmehren) [inline src/logger/console-logger.service.ts:43]: Looking at these statements, I'm wondering why we don't define an order of the levels and just use something like `loglevel <= Loglevel.INFO`

**Review by davidmehren — CHANGES_REQUESTED** [review]

File: src/logger/console-logger.service.ts (hunk 1 above)
Comment (DerMolly) [reply src/logger/console-logger.service.ts]: I build something similar with a helper function…

File: src/config/utils.ts [hunk 2]
```diff
@@ -87,3 +89,27 @@ export function replaceAuthErrorsWithEnvironmentVariables(
   message = message.replace('.accessRole', '_ACCESS_ROLE');
   return message;
 }
+
+export function needToLog(
+  currentLoglevel: Loglevel,
+  requestedLoglevel,
```
Comment (InnayTool) [inline src/config/utils.ts:95]: Is there a reason this is not 
```suggestion
  requestedLoglevel: LogLevel,
```
?

**Review by InnayTool — CHANGES_REQUESTED** [review]

File: src/config/utils.ts (hunk 2 above)
Comment (DerMolly) [reply src/config/utils.ts]: Fixed

**Review by InnayTool — APPROVED** [review]

**Review by davidmehren — APPROVED** [review]

**Pull request merged**